### PR TITLE
add --config (-c) option for specifying exact location of config.json to use.

### PR DESCRIPTION
**Pull request by r-pufky**
Added option per outline in #122 

* Added `--config` (`-c`) option, which accepts a file path to config.json.
* Added logic to verify that both --config and --use-local-config are not used at the same time.
* Updated documentation.

### CODE REVIEW

**Comment by aliparlakci** [comment]
Can you remove the 3d70fb9 commit?

**Comment by r-pufky** [comment]
Sure I'll do it later tonight and update.

**Comment by r-pufky** [comment]
Should be all set. I rebase'd and removed the whitespace cleanup.

File: script.py [hunk 1]
```diff
@@ -256,14 +256,20 @@ def printLogo():
...
+    arguments = Arguments.parse()
+    GLOBAL.arguments = arguments
 
-    if not Path(GLOBAL.defaultConfigDirectory).is_dir():
-        os.makedirs(GLOBAL.defaultConfigDirectory)
-
-    if Path("config.json").exists():
-        GLOBAL.configDirectory = Path("config.json")
+    if arguments.config:
+        if arguments.use_local_config:
```
Comment (aliparlakci) [inline script.py:263]: Why does it exit if `--use-local-config` is given?

File: script.py [hunk 2]
```diff
@@ -256,14 +256,20 @@ def printLogo():
...
+            sys.exit()
+        GLOBAL.configDirectory = Path(arguments.config)
```
Comment (aliparlakci) [inline script.py:265]: There should be a validation that the given file actually exists.

File: src/arguments.py [hunk 3]
```diff
@@ -101,6 +101,7 @@ def parse(arguments=[]):
                             metavar="TIME_LIMIT",
                             type=str)
 
+
```
Comment (aliparlakci) [inline src/arguments.py:104]: Remove this extra line here.

File: script.py (hunk 1 above)
Comment (r-pufky) [reply script.py:263]: These options inherently conflict with each other, with `--use-local-config` creating a new config if it doesn't exist in current directory, as well as loading it from the current directory.

Specifying `--config` implicitly means that you've already created the config or are loading it from a known location.

Happy to remove the check here. I have no strong opinions either way :)

File: src/arguments.py (hunk 3 above)
Comment (r-pufky) [reply src/arguments.py:104]: done.

File: script.py (hunk 2 above)
Comment (r-pufky) [reply script.py:265]: Added validation with log message.

File: script.py (hunk 1 above)
Comment (r-pufky) [reply script.py:263]: Hrmm, it's been two months since your comment with no responses. I'm going to resolve this.

**Pull request merged**